src/hooks/useGame.ts

```diff
@@ -0,0 +1,149 @@
+import { useCallback } from 'react';
+import { TileMeta, Direction } from '../types';
+
+type Position = [number, number];
+
+type MoveStateReturn = [TileMeta[], [TileMeta, TileMeta][], number];
+
+export const getScoreFromMergePairs = (mergePairs: [TileMeta, TileMeta][]): number => {
+  return mergePairs.reduce((acc, item) => (acc || 0) + item[0].value, 0);
+};
+
+export const mergeState = (state: TileMeta[], mergePairs: [TileMeta, TileMeta][]): TileMeta[] => {
+  const mergedState = [...state];
+  const toDeleteTilesIDs: number[] = [];
+
+  for (const [source, destination] of mergePairs) {
+    const sourceIdx = mergedState.findIndex((tile) => tile.id === source.id);
+    mergedState[sourceIdx] = {
+      id: Math.random(),
+      value: source.value * 2,
+      position: destination.position,
+      isMerged: true,
+    };
+    toDeleteTilesIDs.push(destination.id, source.id);
+  }
+
+  return mergedState.filter((tile) => !toDeleteTilesIDs.includes(tile.id));
+};
+
+export const useGame = (size: number) => {
```

Comment:
Move all the functions from `useGame` to global scope by adding `size` parameter to each.